src/main/scala/com/raquo/airstream/core/Observer.scala

```diff
@@ -126,7 +128,7 @@ object Observer {
           if (onTryParam.isDefinedAt(nextValue)) {
             onTryParam(nextValue)
           } else {
-            nextValue.fold(err => AirstreamError.sendUnhandledError(err), identity)
+            nextValue.fold(err => AirstreamError.sendUnhandledError(err), _ => ())
```

Comment:
I couldn't figure out why the success value case is dropped here. Maybe like this:
```
nextValue.failed.foreach { err => 
  AirstreamError.sendUnhandledError(err) 
}
```
?

Reply:
good catch!

Reply:
hmm, `Try,failed` return a `Failure` in case of success. It would not as be clear as `fold` in this case IMHO.

Reply:
Yeah, and because of that foreach will only kick in in the Failed case. It's like with futures:
```scala
val f: Future[String] = ???
f.failed.foreach { case exception => ... }
```

With fold we're supposed to pass in a pure function, but we're dong side-effects and discarding the resulting Try. 

Not insisting, but for me the intent was not clear initially :)

Reply:
`failed` certainly looks nicer, but has a cost I'm not willing to pay in library code. With `failed`, on every incoming (non-error) event we would be initializing and discarding an exception: `Failure(new UnsupportedOperationException("Success.failed"))`. And it seems that this will also [generate a stack trace](https://stackoverflow.com/questions/4195207/are-stack-traces-generated-when-a-java-exception-is-thrown) for that exception, which isn't super cheap.

If the intent of this line is unclear, we could add a comment to clarify, but what should it say? The code itself already says that it's reporting unhandled errors.

Reply:
maybe like this then? :)
```scala
nextValue match {
  case Failure(err) => AirstreamError.sendUnhandledError(err)
  case _ => // nothing to do
}
```

Reply:
Last time I checked, maybe a couple years ago, pattern matches in scala.js were said to be not efficient. Things probably improved since then, but I don't have time to check the generated code.

I guess I'm just not sure what's trippy about this particular `fold`. The codebase is full of `fold`-s, this one having an empty case doesn't seem all that special to me.